documentation/how_to_format_cpp_lambda_functions.rst

```diff
@@ -0,0 +1,76 @@
+#################################
+Formatting C++ Lambda Expressions
+#################################
+
+Uncrustify supports some formatting of c++ lambda expressions, although the support is incomplete.
+A complete description of the c++ lambda expressions 
+The parts of c++ lambda expressions that is currently recognized by Uncrusitify is:
+
+.. code-block:: c++
+
+  [ captures ] execution_context ( params ) specifiers -> ret { body }
+  
+Explanations for all the tokens in the above lambda expression (except for execution_context) are found `here <https://en.cppreference.com/w/cpp/language/lambda>`_.
+The `execution_context` token is a non-standard addition to allow for specification of the execution space (e.g. host or device in CUDA).
+The native specifiers for the `execution_context` for lambda expression in CUDA are ``__device__`` and ``__host__ __device__``.
+However it is common for codes to use a preprocessor variable in place of the native specifiers.
+
+The Uncrustify options for formatting of c++ lambda expressions are as follows:
+
+.. code-block:: c++
```

Comment:
This isn't C++, please omit that.
```suggestion
.. code-block::
```

Also, I'm a little dubious of copying the documentation here. While the added value is obvious, the chance for it to drift out of sync is high. I might replace `s/are as follows/are/` above and then just list the relevant options, either as a (not-C++) `code-block`, or perhaps even as an actual list of literals, i.e.:

```
 - ``sp_cpp_lambda_assign``
 - ``sp_cpp_lambda_square_paren``
 ...
```

(If you do the latter, don't forget a blank line before and after the list, or docutils won't format it properly.)